### PR TITLE
Context inputs lexical ordering

### LINKED ISSUE #1863: ContextInput lexical order
https://github.com/iotaledger/tips/pull/147/files/7cd4554c5f02563a0245b051f0d777c007d5bb2b..8f85f1678d7db958bdd362a0ed9ff333fe7af81f

### PR DESCRIPTION
**Pull request by thibault-martinez**
Closes https://github.com/iotaledger/iota-sdk/issues/1863
https://github.com/iotaledger/tips/pull/147/commits/8f85f1678d7db958bdd362a0ed9ff333fe7af81f

### CODE REVIEW

File: sdk/src/types/block/payload/signed_transaction/transaction.rs [hunk 1]
```diff
@@ -365,38 +369,21 @@ fn verify_context_inputs_packable<const VERIFY: bool>(
...
+            a.account_id().cmp(b.account_id())
         }
+        (ContextInput::Reward(a), ContextInput::Reward(b)) => a.index().cmp(&b.index()),
+        // No need to evaluate all combinations as `then_with` is only called on Equal.
+        _ => unreachable!(),
+    })
+}
+
+fn verify_context_inputs(context_inputs: &[ContextInput]) -> Result<(), Error> {
+    if !is_unique_sorted_by(context_inputs.iter(), |a, b| context_inputs_cmp(a, b)) {
```
Comment (thibault-martinez) [inline sdk/src/types/block/payload/signed_transaction/transaction.rs:385]: Wanted to pass `context_inputs_cmp` directly but there is a double ref issue, didn't have time to investigate much yet

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

File: sdk/src/types/block/payload/signed_transaction/transaction.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/payload/signed_transaction/transaction.rs:385]: @DaughterOfMars Any idea on this?

Comment (DaughterOfMars) [reply sdk/src/types/block/payload/signed_transaction/transaction.rs:385]: I mean you could probably adjust the fn signature of `context_inputs_cmp` but I don't really see the point

Comment (thibault-martinez) [reply sdk/src/types/block/payload/signed_transaction/transaction.rs:385]: But then it makes the sort one to fail compilation. Was just for my rust understanding :)

Comment (DaughterOfMars) [reply sdk/src/types/block/payload/signed_transaction/transaction.rs:385]: ah yeah, well the difference is just that `context_inputs.iter()` and `is_unique_sorted_by` both provide one level of indirection so you end up needing a fn that takes double references.

**Pull request merged**